operator/src/lib.rs

```diff
@@ -0,0 +1,608 @@
+            command.push(adapted_command);
+        }
+
+        let containers = vec![Container {
+            image: Some(image_name),
+            name: "spark".to_string(),
+            // TODO: worker -> add master port
+            command: Some(command),
+            volume_mounts: Some(vec![
+                // One mount for the config directory, this will be relative to the extracted package
+                VolumeMount {
+                    mount_path: "conf".to_string(),
+                    name: "config-volume".to_string(),
+                    ..VolumeMount::default()
+                },
+                VolumeMount {
+                    mount_path: "/tmp".to_string(), // TODO: Make configurable via crd log dir
+                    name: "event-volume".to_string(),
+                    ..VolumeMount::default()
+                },
+            ]),
+            // set no daemonize and configroot via env variables:
+            // if loaded via spark-default.conf or spark-env.sh the variables are set too late!
+            env: Some(vec![
+                EnvVar {
+                    name: "SPARK_NO_DAEMONIZE".to_string(),
+                    value: Some("true".to_string()),
+                    ..EnvVar::default()
+                },
+                EnvVar {
+                    name: "SPARK_CONF_DIR".to_string(),
+                    value: Some("{{configroot}}/conf".to_string()),
+                    ..EnvVar::default()
+                },
+            ]),
+            ..Container::default()
+        }];
+
+        let cm_name = self.create_config_map_name(node_type, hash);
+        let volumes = vec![
+            Volume {
+                name: "config-volume".to_string(),
+                config_map: Some(ConfigMapVolumeSource {
+                    name: Some(cm_name),
+                    ..ConfigMapVolumeSource::default()
+                }),
+                ..Volume::default()
+            },
+            Volume {
+                name: "event-volume".to_string(),
+                ..Volume::default()
+            },
+        ];
+
+        (containers, volumes)
+    }
+
+    async fn create_config_maps(
+        &self,
+        selector: &SparkNodeSelector,
+        node_type: &SparkNodeType,
+        hash: &str,
+    ) -> Result<(), Error> {
+        // TODO: remove hardcoded and make configurable and distinguish master / worker vs history-server
+        let mut config_properties: HashMap<String, String> = HashMap::new();
+        config_properties.insert(
+            "spark.history.fs.logDirectory".to_string(),
+            "/tmp".to_string(),
+        );
+
+        config_properties.insert("spark.eventLog.enabled".to_string(), "true".to_string());
+
+        config_properties.insert("spark.eventLog.dir".to_string(), "/tmp".to_string());
+        let mut env_vars: HashMap<String, String> = HashMap::new();
+        // TODO: use product-conf for validation
+
+        let mut handlebars_config = Handlebars::new();
```

Comment:
Nothing for you to do but I don't like the handlebar template stuff in our code.
I'm sure we can find a _nicer_ way and/or move some of it to operator-rs

Reply:
Yeah i do not like it either. It can easily be wrapped in simple methods (which maybe outsourced to product-config?)